src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -123,7 +99,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Target Name="_CheckForUnsupportedTargetPlatform"
           BeforeTargets="_CheckForInvalidConfigurationAndPlatform;RunResolvePackageDependencies;GetFrameworkPaths;GetReferenceAssemblyPaths"
-          Condition="'$(_InferredTargetPlatform)' == 'true'" >
+          Condition="'$(_InferredTargetPlatform)' == 'true' and '$(ErrorOnUnsupportedTargetPlatform)' != 'false'" >
```

Comment:
It will give the same error? Why extra flag on `ErrorOnUnsupportedTargetPlatform` is needed?

Reply:
I was giving an escape hatch for projects like [this test asset](https://github.com/dotnet/sdk/pull/12202/files#diff-7dde177f90cb7a97b30bfe7b4aad8d44R14), which does some custom parsing of the target framework `net40-client`. 
I'm not sure the behavior we want here, right now it errors when the new target framework format (`{target framework}-{target platform}`) is used with a target framework below .NET 5. An alternative is to not attempt to parse out the target platform at all if the framework is below .NET 5, so it wouldn't error in cases like `netcoreapp3.1-ios` but it would just ignore the ios part of the target framework.

Reply:
@dsplaisted what do you think? I think error even it is below netcoreapp3.1 is fine, giving the escape hatch. @sfoslund you should mention the escape hatch in the error message tho. Something the guide the user to set the property.

Reply:
Great, I just added the logic for #11232 to this PR.